src/server/getters/get-dispute-info.ts

```diff
@@ -1,77 +1,117 @@
+    disputeRound: (next: AsyncCallback) => db("crowdsourcers").select("marketID").count("* as disputeRound").groupBy("crowdsourcers.marketID").where("crowdsourcers.completed", 1).whereIn("crowdsourcers.marketID", marketIDs).asCallback(next),
   }, (err: Error|null, stakeResults: DisputesResult): void => {
     if (err) return callback(err);
     if (!stakeResults.markets) return callback(new Error("Could not retrieve markets"));
 
     const marketDisputeDetails: Array<DisputesResult> =
       _.map(marketIDs, (marketID: Address) =>
-        _.mapValues(stakeResults, (result) => _.filter(result, {marketID})),
+        _.mapValues(stakeResults, (result) => _.filter(result, { marketID })),
       );
     callback(null, _.map(marketDisputeDetails, reshapeStakeRowToUIStakeInfo));
   });
 }
 
 function reshapeStakeRowToUIStakeInfo(stakeRows: DisputesResult): UIStakeInfo|null {
-  if (stakeRows.markets.length === 0) {
+  const marketRow = stakeRows.markets[0];
+  if (marketRow == null) {
     return null;
   }
-  const payoutGrouped: { [payoutID: number]: Array<StakeRow>} = _.groupBy(stakeRows.crowdsourcers.concat(stakeRows.initialReport), "payoutID");
-  const stakeResults = _.map(payoutGrouped, (stakes: Array<StakeRow>) => {
-    const totalStaked = _.sumBy(stakes, (stake: StakeRow) => {
-      return stake.amountStaked;
-    });
-    const activeCrowdsourcer = _.find(stakes, (stake: StakeRow) => stake.completed === null);
-    let size: number;
-    let amountStaked: number;
-    if (activeCrowdsourcer == null) {
-      // TODO: calculate from existing stakes if no crowdsourcer created yet
-      size = 0;
-      amountStaked = 0;
+  const totalCompletedStakeOnAllPayouts = new BigNumber(
+    _.sum(_.map(stakeRows.completedStakes, (completedStake) => completedStake.amountStaked)));
+
+  const completedStakeByPayout: { [payoutID: number]: StakeRow } = _.keyBy(stakeRows.completedStakes, "payoutID");
+  const activeCrowdsourcerByPayout: { [payoutID: number]: ActiveCrowdsourcer } = _.keyBy(stakeRows.activeCrowdsourcer, "payoutID");
+
+  const stakeResults = _.map(stakeRows.payouts, (payout: PayoutRow) => {
+    const completedStakes = completedStakeByPayout[payout.payoutID];
+    const activeCrowdsourcer = activeCrowdsourcerByPayout[payout.payoutID];
+
+    const completedStakeAmount = new BigNumber(completedStakes == null ? 0 : completedStakes.amountStaked);
+    const totalStakeOnPayout = completedStakeAmount.add(new BigNumber(activeCrowdsourcer == null ? 0 : activeCrowdsourcer.amountStaked));
+
+    let currentAmounts: { size?: string; currentStake?: string };
+    if (payout.tentativeWinning === 1 || !isActiveMarketState(marketRow.reportingState)) {
+      currentAmounts = {};
+    } else if (activeCrowdsourcer == null) {
```

Comment:
if there is a tentativeWinning I would think there is a current amount staked. tentative winning should be set after initial reporter. Is it just the designated reporter stake? if only DR reported.

Reply:
This branch here only provides the size (for current dispute) and currentStake (on that dispute). totalStakeOnPayout is calculated above, and for every payout